AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/helper/RestAnnotationHelper.java

```diff
@@ -359,7 +375,18 @@ private JClass resolveResponseClass(TypeMirror expectedType, RestHolder holder)
 			return decoratedExpectedClass;
 		} else if (expectedType.getKind() == TypeKind.ARRAY) {
 			ArrayType arrayType = (ArrayType) expectedType;
-			return resolveResponseClass(arrayType.getComponentType(), holder).array();
+
+			TypeMirror componentType = arrayType.getComponentType();
+
+			if (componentType.getKind() == TypeKind.DECLARED) {
+				DeclaredType declaredType = (DeclaredType) componentType;
+
+				if (useTypeReference && getElementUtils().getTypeElement(RestSpringClasses.PARAMETERIZED_TYPE_REFERENCE) != null) {
+					return codeModelHelper.typeMirrorToJClass(expectedType);
```

Comment:
shouldnt the `expectedType` be `declaredType`? if not you should remove the unused variable `declaredType`.